VMEncryption/main/DiskUtil.py

```diff
@@ -1011,20 +1011,57 @@ def should_skip_for_inplace_encryption(self, device_item, special_azure_devices_
             return False
 
     def get_azure_devices(self):
-        ide_devices = self.get_ide_devices()
+        device_names = []
         blk_items = []
+
+        # Get all IDE devices
+        ide_devices = self.get_ide_devices()
         for ide_device in ide_devices:
-            current_blk_items = self.get_device_items("/dev/" + ide_device)
-            for current_blk_item in current_blk_items:
-                blk_items.append(current_blk_item)
+            device_names.append("/dev/" + ide_device)
 
-        scsi0_devices = self.get_scsi0_device_names()
-        for scsi0_device in scsi0_devices:
-            current_blk_items = self.get_device_items(scsi0_device)
+        # get all SCSI 0 devices
+        device_names += self.get_scsi0_device_names()
+
+        # some machines use special root dir symlinks instead of scsi0 symlinks
+        device_names += self.get_azure_symlinks_root_dir_devices()
+
+        # let us do some de-duping
+        device_names_realpaths = set(map(os.path.realpath, device_names))
+
+        for device_path in device_names_realpaths:
+            current_blk_items = self.get_device_items(device_path)
             for current_blk_item in current_blk_items:
                 blk_items.append(current_blk_item)
+
         return blk_items
 
+    def get_azure_symlinks_root_dir_devices(self):
```

Comment:
Can we unit test this?  Mocks might be a pain, but would be nice to cover if possible.